### PR TITLE
UHM-4880, support installing PatientIdentifierType with UniquenessBehavior

### PR DESCRIPTION
**Pull request by cioan**
@mogoodrich , after talking to you in this morning, I kept working on adding UniquenessBehavior and I got it working. It does not look right to me, I think this code could be simplified and improved? I will send you a separate PR for the changes I had to make in the pihcore module. Let me know what you think. Thanks!

### CODE REVIEW

**Review by mogoodrich — COMMENTED** [review]
Thanks @cioan !  If it actually works, the code looks good to me... all the tests pass as well?

@dkayiwa  could you take a look at this, I'm not an expert at all in how Core version-specific submodules work... also @mseaton ?

**Comment by cioan** [comment]
Thanks @mogoodrich ! Yes, the current tests are passing. I could also add tests for the new 1_11 module I created, but first just wanted to make sure the packages and classes I created are correct.

**Comment by dkayiwa** [comment]
Where do i find the JIRA ticket that is in the commit message?

**Comment by cioan** [comment]
Sorry @dkayiwa , this is a PIH JIRA ticket. I will create an OpenMRS ticket. The requirement is to be able to install PatientIdentifierType with UniquenessBehavior via the metadatadeploy. The PatientIdentifierType.UniquenessBehavior is available from openmrs versions 1.11 and greater. Thanks!

**Comment by cioan** [comment]
@dkayiwa , here is the OpenMRS ticket for this new feature:
https://issues.openmrs.org/browse/DPLY-52
Thanks!

**Comment by dkayiwa** [comment]
I have made it ready for work.

**Comment by cioan** [comment]
Thanks @dkayiwa ! Does the approach I implemented look reasonable to you?

File: api-1.11/src/main/java/org/openmrs/module/metadatadeploy/bundle/AbstractMetadataBundle1_11.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.openmrs.module.metadatadeploy.bundle;
+
+import org.openmrs.annotation.OpenmrsProfile;
+import org.openmrs.module.metadatadeploy.descriptor.PatientIdentifierTypeDescriptor1_11;
+
+@OpenmrsProfile(openmrsVersion = "1.11.*")
```
Comment (dkayiwa) [inline api-1.11/src/main/java/org/openmrs/module/metadatadeploy/bundle/AbstractMetadataBundle1_11.java:6]: Did you mean to use `1.11.0`?

Comment (cioan) [reply api-1.11/src/main/java/org/openmrs/module/metadatadeploy/bundle/AbstractMetadataBundle1_11.java:6]: Sure, being honest, I do not know when to use 1.11.0 and when to use 1.11.*. I just searched through openmrs code and I saw in other places was used 1.11.*

Comment (dkayiwa) [reply api-1.11/src/main/java/org/openmrs/module/metadatadeploy/bundle/AbstractMetadataBundle1_11.java:6]: Is this of help? https://github.com/openmrs/openmrs-core/blob/2.3.1/api/src/main/java/org/openmrs/module/ModuleUtil.java#L256-L261

Comment (cioan) [reply api-1.11/src/main/java/org/openmrs/module/metadatadeploy/bundle/AbstractMetadataBundle1_11.java:6]: That makes a lot of sense! Then, for sure I meant 1.11.0. Thanks @dkayiwa !

**Comment by dkayiwa** [comment]
@cioan feel free to claim the JIRA ticket.

**Review by mogoodrich — APPROVED** [review]
Thanks for the review @dkayiwa ... looks good to me @cioan once we update the commit message (which we can do when we merge).

**Comment by cioan** [comment]
Thanks @dkayiwa and @mogoodrich ! I just a couple unit tests. Thanks!

**Pull request merged**